editoast/migrations/2024-02-21-002011_create_v2_scenario_trainschedule_timetable/up.sql

```diff
@@ -0,0 +1,32 @@
+CREATE TABLE timetablev2 (
+    id int8 PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    electrical_profile_set_id int8 NULL REFERENCES electrical_profile_set(id) ON DELETE CASCADE
+);
+CREATE TABLE scenariov2 (
+    id int8 PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    infra_id int8 NOT NULL REFERENCES infra(id) ON DELETE CASCADE DEFERRABLE INITIALLY DEFERRED,
+    name varchar(128) NOT NULL,
+    description varchar(1024) NOT NULL,
+    creation_date timestamptz NOT NULL,
+    last_modification timestamptz NOT NULL,
+    tags text [] NOT NULL,
+    timetable_id int8 NOT NULL UNIQUE REFERENCES timetablev2(id) DEFERRABLE INITIALLY DEFERRED,
+    study_id int8 NOT NULL REFERENCES study(id) ON DELETE CASCADE
+);
+CREATE TABLE trainschedulev2 (
+    id int8 PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
+    train_name varchar(128) NOT NULL,
+    labels text [] NOT NULL,
+    rolling_stock_name varchar(128) NOT NULL,
+    timetable_id int8 NOT NULL UNIQUE REFERENCES timetablev2(id) DEFERRABLE INITIALLY DEFERRED,
```

Comment:
- It should not be unique otherwise we can't have two train schedules attached to the same timetable.
- It should DROP CASCADE in case of deletion of the timetable.
- I don't see the use case of the `DEFERRABLE INITIALLY DEFERRED`